resource/object.go

```diff
@@ -149,16 +150,141 @@ type CommonMetadata struct {
 //	    return resource.CopyObject(c)
 //	}
 func CopyObject(in any) Object {
```

Comment:
Should this be renamed to `DeepCopyObject` so we are clear?  (and keep a CopyObject that is not deep?)

Reply:
I'm not sure we need/want a CopyObject that isn't deep, or rather, if we do, we might want it to be clearer from the naming that it's a shallow copy.